### ISSUE TITLE
Using condition/s software not using Released state.

### ISSUE THREAD

**Issue by Forsaken-Player** [WhiteMagic/JoystickGremlin#214]
When I try to use Release state in a condition, that condition doesn't work.

**Comment by WhiteMagic** [comment]
I assume this is happening with Remap, Map to Keyboard, and Map to Mouse actions but not Macro, Text to Speech, etc?

The reason I split them is that the first three all do something both when a physical input is pressed and later on released while the other actions all typically only do something when the physical input is pressed. The fact that they run on both press and release events makes them odd in many cases and I will have to check what / if any crazy logic exists in the condition code to handle those oddball actions. The tricky part most likely will be to figure out what the "correct" thing to do for those actions is when paired with a condition which effectively only dictates when something runs but not when it stops.

A workaround until that time is to use macros that emit the, for example vjoy button press and release events as desired.

**Comment by Forsaken-Player** [comment]
Hello. I fact I could not make any condition to work if I set in release with both physical input and virtual. Tried macros and remap.

No way.

I'll keep testing tomorrow.

**Comment by willxy** [comment]
This sounds similar to what I'm observing.  When I set a Condition to eg "Activate if vJoy 1 Button 64 is Released", the Condition seems to keep reverting to "Pressed" instead.  The reverting seems to happen when I load the profile, or when I change the profile from Activated to Deactivated.

It does seem to save to the xml file properly, if I save it before it seems to revert:
```
<activation-condition rule="all">
    <condition comparison="released" condition-type="vjoy" id="64" input="button" vjoy-id="1"/>
</activation-condition>
```
And of course if I save it after it reverts, it's reverted in the xml file as well:
```
<activation-condition rule="all">
    <condition comparison="pressed" condition-type="vjoy" id="64" input="button" vjoy-id="1"/>
</activation-condition>
```

**Comment by WhiteMagic** [comment]
This version seems to fix that issue https://github.com/WhiteMagic/JoystickGremlin/releases/tag/Release_13.3_debug. When testing the resetting of condition settings the above version doesn't seem to cause the problems encountered with 13.3

**Comment by wakirk** [comment]
This issue is still occurring.   I confirmed when loading a profile with:
```
                                <activation-condition rule="all">
                                    <condition comparison="pressed" condition-type="vjoy" id="1" input="button" vjoy-id="4"/>
                                    <condition comparison="released" condition-type="vjoy" id="2" input="button" vjoy-id="4"/>
                                    <condition comparison="released" condition-type="vjoy" id="3" input="button" vjoy-id="4"/>
                                    <condition comparison="released" condition-type="vjoy" id="4" input="button" vjoy-id="4"/>
                                </activation-condition>
```
All 4 conditions loaded as 'pressed' into the UI.  Also, after setting the value and activating the profile, they revert to 'pressed' state.  

About window reports version 13.3.

A copy of this profile was created with a Thrustmaster TWCS Throttle (paired with a T16000M) 

The full copy can be found here: https://drive.google.com/file/d/1Vuz1iXKMEtXfFTFCkiUm04x88WdAZicR/view?usp=sharing

**Comment by kschaab** [comment]
Yeah I'm seeing this too, if you set the condition to released and activate it works but as soon as you reload the profile it goes backed to pressed.  Before reload:

![image](https://user-images.githubusercontent.com/7329029/210129183-4539ecb1-8811-43d6-9a67-0adbfb011c7a.png)

After reload:
![image](https://user-images.githubusercontent.com/7329029/210129202-a1b9e981-915f-4662-bd3b-3fb6612fbc79.png)

Functionality mimics this condition.

**Comment by WhiteMagic** [comment]
The UI in R14 for conditions is entirely different such that this is no longer an issue.
